src/shims/alloc.rs

```diff
@@ -38,6 +38,9 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         }
         // We have `size < min_align`. Round `size` *down* to the next power of two and use that.
         fn prev_power_of_two(x: u64) -> u64 {
+            if x == 0 {
+                return 1;
+            }
```

Comment:
I wonder if it is better to do it outside ``prev_power_of_two`` as 0 is not the previous power of two.